include/frozen/bits/pmh.h

```diff
@@ -152,15 +148,29 @@ struct seed_or_index {
 
 // Represents the perfect hash function created by pmh algorithm
 template <std::size_t M, class Hasher>
-struct pmh_tables {
+struct pmh_tables : private Hasher {
```

Comment:
Any reason to prefer private inheritance over membership?

Reply:
Yes, a very specific reason: the [empty-base-optimization].

I described this in my commit message for cb928e0ddded95e7036a223c34574a66bccf45da, but unfortunately GitHub's UI makes commit messages difficult to spot in PRs, so let me copy it here:
>  improvement: allow empty key comparison and hash functors to occupy no space
> 
> This relies on the [empty-base-optimization] to ensure these occupy no
> space when they're empty. As member variables unique objects are
> required to have unique addresses, effectively mandating at least 1 byte
> plus potential padding. This restriction does not apply to base classes
> which are allowed to share their address with their derived class.

From C++20 onwards it would be possible to use the [`[[no_unique_address]]`](https://en.cppreference.com/w/cpp/language/attributes/no_unique_address) attribute for the same purpose instead, but that's a long way away from C++14. (Also using it on C++20 only would result in a differing ABI for this type before and after C++20. I like having stable ABIs, not just APIs, wherever possible).

[empty-base-optimization]: https://en.cppreference.com/w/cpp/language/ebo